packages/wrangler/src/assets.ts

```diff
@@ -538,18 +519,67 @@ export function validateAssetsArgsAndConfig(
 
 	// Smart placement turned on when using assets
 	if (
-		config?.placement?.mode === "smart" &&
-		config?.assets?.run_worker_first === true
+		config.placement?.mode === "smart" &&
+		config.assets?.run_worker_first === true
 	) {
 		logger.warn(
 			"Turning on Smart Placement in a Worker that is using assets and run_worker_first set to true means that your entire Worker could be moved to run closer to your data source, and all requests will go to that Worker before serving assets.\n" +
 				"This could result in poor performance as round trip times could increase when serving assets.\n\n" +
 				"Read more: https://developers.cloudflare.com/workers/static-assets/binding/#smart-placement"
 		);
 	}
+
+	if (
+		(args.assets || config.assets?.directory) &&
+		!(args.script || config.main)
+	) {
+		const unsupportedConfigKeys = getConfigKeysUnsupportedByAssetsOnly(config);
+
+		if (unsupportedConfigKeys.length > 0) {
+			const keys = unsupportedConfigKeys.map((key) => `⋅ "${key}"`).join("\n");
+
+			throw new UserError(
+				`Assets-only Workers do not support the following configuration keys:\n\n` +
+					`${keys}\n\n` +
+					`Please remove these fields from your configuration file, or configure the "main" field if you are trying to deploy a Worker with assets.`
+			);
+		}
+	}
 }
 
-const WORKER_JS_FILENAME = "_worker.js";
+function getConfigKeysUnsupportedByAssetsOnly(config: Config): Array<string> {
+	const supportedAssetsOnlyConfigKeys = new Set([
+		"name",
+		"compatibility_date",
+		"compatibility_flags",
+		"assets",
+		"build",
```

Comment:
why is build supported?

Reply:
build being supported feels very reasonable to me. it's a Wrangler-only feature (doesn't make it to EWC), and would be useful if you want to build a static site, for example.